sklearn/utils/_docstring.py

```diff
@@ -0,0 +1,26 @@
+"""Docstring utilities."""
+
+
+class Substitution:
```

Comment:
can we just call this `inject_docstring`? Or a name that makes it clear the substitution is about docstrings.

Also to avoid the uppercase we could do:

```py
def inject_docstring(...):
	return _InjectDocstring(...)

class _InjectDcostring:
	...
```

Reply:
I went for a decorator with parameters instead of using the class.